overwatch-rs/src/services/status.rs

```diff
@@ -0,0 +1,98 @@
+// std
+use std::default::Default;
+use std::marker::PhantomData;
+use std::sync::Arc;
+use std::time::Duration;
+// crates
+use crate::services::{ServiceData, ServiceId};
+use thiserror::Error;
+use tokio::sync::watch;
```

Comment:
The reason why we use `tokio::sync::watch` is to allow multiple services to subscribe the status of a target service when they really want? In other words, to prevent the value inside the channel from being consumed until the target service produces a new status?

Reply:
watch is a channel with a single item that gets updated (only last updated value is kept, guaranteed). Sender is not clone, but just need and &Sender to update the value, so it's easily shared wrapped in Arc